<?xml version="1.0" ?>

<launch>

    <arg name="urdfFile" default="$(find qm_description)/urdf/quadruped_manipulator/robot.urdf"/>
    <arg name="taskFile" default="$(find qm_controllers)/config/task.info"/>
    <arg name="referenceFile" default="$(find qm_controllers)/config/reference.info"/>
    <arg name="gaitCommandFile" default="$(find qm_controllers)/config/gait.info"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="$(arg urdfFile)"/>
    <param name="taskFile" value="$(arg taskFile)"/>
    <param name="referenceFile" value="$(arg referenceFile)"/>
    <param name="gaitCommandFile" value="$(arg gaitCommandFile)"/>

    <rosparam file="$(find qm_controllers)/config/controllers.yaml" command="load"/>

    <node name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/qm_controller
    "/>
</launch>
